File: C++/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P02_ExamPreparation.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

	string input;
	getline(cin, input);
	int numPoorGrades = stoi(input);
	string command;
	getline(cin, command);
	string secondInput;
	int currGrade;
	int counterSubjects = 0;
	string lastSubject = command;
	int counterPoorGrades = 0;
	int totalGrades = 0;

	while (command != "Enough") {
		counterSubjects++;
		getline(cin, secondInput);
		currGrade = stoi(secondInput);
		totalGrades += currGrade;
		lastSubject = command;
		if (currGrade <= 4) {
			counterPoorGrades++;
		}

		if (counterPoorGrades == numPoorGrades) {
			cout << "You need a break, " << numPoorGrades << " poor grades." << endl;

			return 0;
		}

		getline(cin, command);
	}

	cout.setf(ios::fixed);
	cout.precision(2);

	double averageGrade = static_cast<double>(totalGrades) / counterSubjects;

	cout << "Average score: " << averageGrade << endl;
	cout << "Number of problems: " << counterSubjects << endl;
	cout << "Last problem: " << lastSubject << endl;

	return 0;
}

File: Python/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P02_ExamPreparation.py
max_bad_scores = int(input())
bad_scores = 0
tasks = 0
total_score = 0
last_problem = 0

while True:
    name_task = input()

    if name_task == "Enough":
        print(f"Average score: {total_score / tasks:.2f}")
        print(f"Number of problems: {tasks}")
        print(f"Last problem: {last_problem}")
        break

    score = int(input())
    tasks += 1
    total_score += score
    last_problem = name_task

    if score <= 4:
        bad_scores += 1

    if bad_scores == max_bad_scores:
        print(f"You need a break, {bad_scores} poor grades.")
        break
